src/zeal/docset.py

```diff
@@ -1,28 +1,30 @@
 import os
+import platform
 import shutil
+import subprocess
 import urllib
+from pathlib import Path
 
 import bs4
 
 from . import config, downloads, exceptions
 
 LATEST_VERSION = "latest"
 
-def list_all(docset_dir: str = config.docset_dir) -> list:
+def list_all(docset_dir: Path = config.docset_dir) -> list:
     """List the docsets in the docset_dir.
 
     :param docset_dir: String, path to the Zeal docset directory. DefaultL filesystem.docset_dir
     :return: List of docsets (by name, not by path)
     """
-    files_list = os.listdir(docset_dir)
     installed_docsets = []
-    for file in files_list:
-        if os.path.isdir(os.path.join(docset_dir, file)) and file.endswith(".docset"):
-            installed_docsets.append(file.removesuffix(".docset"))
+    for elem in docset_dir.glob("*.docset"):
+        if elem.is_dir():
+            installed_docsets.append(elem.stem)
     return installed_docsets
 
 
-def download(docset_name: str, feeds_dir: str, docset_version: str = LATEST_VERSION, docset_dir: str = config.docset_dir) -> None:
+def download(docset_name: str, feeds_dir: Path, docset_version: str = LATEST_VERSION, docset_dir: Path = config.docset_dir) -> None:
     """Download a docset by its feed name.
 
     :param docset_name: String, the feed name of the docset to download
```

Comment:
Line 31: param `feeds_dir` is no longer a string, should be `pathlib.Path`
Line 33: param `docset_dir` is no longer a string, should be `pathlib.Path`